demo/jvm/src/main/scala/games/Utils.scala

```diff
@@ -0,0 +1,184 @@
+      }
+    }
+  }
+
+  /**
+   * Add a Runnable task to be executed by the OpenGL thread.
+   * Tasks are usually executed at the beginning of the next iteration of the FrameListener loop system.
+   */
+  def addPendingTask(task: Runnable): Unit = {
+    pendingTaskList.add(task)
+  }
+
+  def streamForResource(res: Resource): InputStream = {
+    val stream = JvmUtils.getClass().getResourceAsStream(res.name)
+    if (stream == null) throw new RuntimeException("Could not load resource " + res.name)
+    stream
+  }
+}
+
+trait UtilsImpl extends UtilsRequirements {
+  def getBinaryDataFromResource(res: games.Resource)(implicit ec: ExecutionContext): scala.concurrent.Future[java.nio.ByteBuffer] = {
+    Future {
+      val stream = JvmUtils.streamForResource(res)
+      val byteStream = new ByteArrayOutputStream()
+      val tmpData: Array[Byte] = new Array[Byte](4096) // 4KiB of temp data
+
+      var tmpDataContentSize: Int = 0
+      while ({ tmpDataContentSize = stream.read(tmpData); tmpDataContentSize } >= 0) {
+        byteStream.write(tmpData, 0, tmpDataContentSize)
+      }
+
+      stream.close()
+
+      val byteArray = byteStream.toByteArray()
+      val byteBuffer = ByteBuffer.allocate(byteArray.length)
+
+      byteBuffer.put(byteArray)
+      byteBuffer.rewind()
+
+      byteBuffer
+    }
+  }
+  def getTextDataFromResource(res: games.Resource)(implicit ec: ExecutionContext): scala.concurrent.Future[String] = {
+    Future {
+      val stream = JvmUtils.streamForResource(res)
+      val streamReader = new InputStreamReader(stream)
+      val reader = new BufferedReader(streamReader)
+
+      val text = new StringBuilder()
+
+      val buffer = Array[Char](4096) // 4KiB buffer
+      var bufferReadLength = 0
+
+      while ({ bufferReadLength = reader.read(buffer); bufferReadLength } >= 0) {
+        text.appendAll(buffer, 0, bufferReadLength)
+      }
+
+      reader.close()
+      streamReader.close()
+      stream.close()
+
+      text.toString()
+    }
+  }
+  def loadTexture2DFromResource(res: games.Resource, texture: games.opengl.Token.Texture, preload: => Boolean = true)(implicit gl: games.opengl.GLES2, ec: ExecutionContext): scala.concurrent.Future[Unit] = {
+    Future {
+      val stream = JvmUtils.streamForResource(res)
+
+      // Should support JPEG, PNG, BMP, WBMP and GIF
+      val image = ImageIO.read(stream)
+
+      val height = image.getHeight()
+      val width = image.getWidth()
+      val byteBuffer = GLES2.createByteBuffer(4 * width * height) // Stored as RGBA value: 4 bytes per pixel
+      val tmp = new Array[Byte](4)
+      var y = 0
+      while (y < height) {
+        var x = 0
+        while (x < width) {
+          val argb = image.getRGB(x, y)
+          tmp(2) = argb.toByte // blue
+          tmp(1) = (argb >> 8).toByte // green
+          tmp(0) = (argb >> 16).toByte // red
+          tmp(3) = (argb >> 24).toByte // alpha
+          byteBuffer.put(tmp)
```

Comment:
Why don't you get an `IntBuffer` view of the `byteBuffer` in `BigEndian` order:
```scala
val intBuffer = byteBuffer.duplicate().order(ByteOrder.BIG_ENDIAN).asIntBuffer()
```
and then simply
```scala
val argb = image.getRGB(x, y)
intBuffer.put((argb >>> 24) | (argb << 8))
```
?